crates/router/src/core/payment_methods.rs

```diff
@@ -1324,62 +1326,73 @@ pub async fn vault_payment_method(
 ))]
 async fn get_pm_list_context(
-    state: &SessionState,
-    payment_method: &enums::PaymentMethod,
-    _key_store: &domain::MerchantKeyStore,
-    pm: &domain::PaymentMethod,
-    _parent_payment_method_token: Option<String>,
+    payment_method_type: enums::PaymentMethod,
+    payment_method: &domain::PaymentMethod,
     is_payment_associated: bool,
 ) -> Result<Option<PaymentMethodListContext>, error_stack::Report<errors::ApiErrorResponse>> {
-    let payment_method_retrieval_context = match payment_method {
-        enums::PaymentMethod::Card => {
-            let card_details = cards::get_card_details_with_locker_fallback(pm, state).await?;
-
-            card_details.as_ref().map(|card| PaymentMethodListContext {
-                card_details: Some(card.clone()),
-                #[cfg(feature = "payouts")]
-                bank_transfer_details: None,
+    let payment_method_data = payment_method
+        .payment_method_data
+        .clone()
+        .map(|payment_method_data| payment_method_data.into_inner().expose().into_inner());
+
+    let payment_method_retrieval_context = match payment_method_data {
+        Some(payment_methods::PaymentMethodsData::Card(card)) => {
+            Some(PaymentMethodListContext::Card {
+                card_details: api::CardDetailFromLocker::from(card),
                 hyperswitch_token_data: is_payment_associated.then_some(
                     storage::PaymentTokenData::permanent_card(
-                        Some(pm.get_id().clone()),
-                        pm.locker_id
+                        Some(payment_method.get_id().clone()),
+                        payment_method
+                            .locker_id
                             .as_ref()
-                            .map(|id| id.get_string_repr().clone())
-                            .or(Some(pm.get_id().get_string_repr().to_owned())),
-                        pm.locker_id
+                            .map(|id| id.get_string_repr().to_owned())
+                            .or_else(|| Some(payment_method.get_id().get_string_repr().to_owned())),
+                        payment_method
+                            .locker_id
                             .as_ref()
-                            .map(|id| id.get_string_repr().clone())
-                            .unwrap_or(pm.get_id().get_string_repr().to_owned()),
+                            .map(|id| id.get_string_repr().to_owned())
+                            .unwrap_or_else(|| {
+                                payment_method.get_id().get_string_repr().to_owned()
+                            }),
                     ),
                 ),
             })
         }
+        Some(payment_methods::PaymentMethodsData::BankDetails(bank_details)) => {
+            let get_bank_account_token_data =
+                || -> errors::CustomResult<payment_methods::BankAccountTokenData, errors::ApiErrorResponse> {
+                    let connector_details = bank_details
```

Comment:
why we need to take first here?

Reply:
This seems to be related to payment method authentication flows.

Looping in @Sarthak1799.

Reply:
The types and flow here seem specific to PM auth flows in v1.
The first connector details are stored in redis for payment token flow (This would later on be fetched in payments confirm).